src/Ocelot/Metadata/DownstreamRouteExtensions.cs

```diff
@@ -0,0 +1,164 @@
+        new(StringComparer.OrdinalIgnoreCase)
+        {
+            "true",
+            "yes",
+            "on",
+            "ok",
+            "enable",
+            "enabled",
+            "1",
+        };
+
+    /// <summary>
+    /// The known falsy values
+    /// </summary>
+    private static readonly HashSet<string> FalsyValues =
+        new(StringComparer.OrdinalIgnoreCase)
+        {
+            "false",
+            "no",
+            "off",
+            "disable",
+            "disabled",
+            "0",
+        };
+
+    /// <summary>
+    /// The known numeric types
+    /// </summary>
+    private static readonly HashSet<Type> NumericTypes = new()
+    {
+        typeof(byte),
+        typeof(sbyte),
+        typeof(short),
+        typeof(ushort),
+        typeof(int),
+        typeof(uint),
+        typeof(long),
+        typeof(ulong),
+        typeof(float),
+        typeof(double),
+        typeof(decimal),
+    };
+
+    /// <summary>
+    /// Extension method to get metadata from a downstream route.
+    /// </summary>
+    /// <typeparam name="T">The metadata target type.</typeparam>
+    /// <param name="downstreamRoute">The current downstream route.</param>
+    /// <param name="key">The metadata key in downstream route Metadata dictionary.</param>
+    /// <param name="defaultValue">The fallback value if no value found.</param>
+    /// <param name="jsonSerializerOptions">Custom json serializer options if needed.</param>
+    /// <returns>The parsed metadata value.</returns>
+    public static T GetMetadata<T>(this DownstreamRoute downstreamRoute, string key, T defaultValue = default,
+        JsonSerializerOptions jsonSerializerOptions = null)
+    {
+        var metadata = downstreamRoute?.MetadataOptions.Metadata;
+
+        if (metadata == null || !metadata.TryGetValue(key, out var metadataValue) || metadataValue == null)
+        {
+            return defaultValue;
+        }
+
+        return (T)ConvertTo(typeof(T), metadataValue, downstreamRoute.MetadataOptions,
+            jsonSerializerOptions ?? new JsonSerializerOptions(JsonSerializerDefaults.Web));
+    }
+
+    /// <summary>
+    /// Converting a string value to the target type
+    /// Some custom conversion has been for the following types:
+    /// bool, bool?, string[], numeric types
+    /// otherwise trying to deserialize the value using the JsonSerializer
+    /// </summary>
+    /// <param name="targetType">The target type.</param>
+    /// <param name="value">The string value.</param>
+    /// <param name="metadataOptions">The metadata options, it includes the global configuration.</param>
+    /// <param name="jsonSerializerOptions">If needed, some custom json serializer options.</param>
+    /// <returns>The converted string.</returns>
+    private static object ConvertTo(Type targetType, string value, MetadataOptions metadataOptions,
+        JsonSerializerOptions jsonSerializerOptions)
```

Comment:
Because this is private method we can easily redefine it as a generic method.
```suggestion
    private static T ConvertTo<T>(string value, MetadataOptions metadataOptions, JsonSerializerOptions jsonSerializerOptions)
```

Reply:
Does it work like that? I haven't tried it

Reply:
In the .NET framework, conversions typically involve two methods:
- A standard method that returns the most common .NET framework `object` type, which is a boxed value if the type is a value type.
- A generic method that returns a value of the specified type, abstracting from `Type` in favor of type parameters.

The first method is generally straightforward to implement, while the second method can be more complex.
The decision to not include a generic method version is yours to make.
However, we can consider adding a generic version during the next refactoring phase of this logic.
For the moment, it is not an issue.